src/resty/gcp/request/credentials/accesstoken.lua

```diff
@@ -74,6 +74,11 @@ local function GetAccessTokenBySA(serviceAccount)
         return
     end
     local jwtToken = GetJwtToken(serviceAccount)
+    local res = assert(GetAccessTokenByJwt(jwtToken))
+    if res.error then
+        ngx.log(ngx.ERR, "[accesstoken] Unable to get access token: ", res.error_description)
+        return
+    end
```

Comment:
This logic seems odd to me. If no error is encountered `GetAccessTokenByJwt(jwtToken)` will be called twice. Once on line 77, and a second time on line 82.

Reply:
A reminder of debugging code.